packages/lesswrong/components/review/ReviewsLeaderboard.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from 'react';
+import { registerComponent, Components } from '../../lib/vulcan-lib';
+import groupBy from 'lodash/groupBy';
+import sortBy from 'lodash/sortBy';
+import Card from '@material-ui/core/Card';
+import { ReviewYear } from '../../lib/reviewUtils';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: {
+    ...theme.typography.body2,
+    color: theme.palette.grey[600],
+    background: theme.palette.background.pageActiveAreaBackground,
+    padding: 16,
+    paddingBottom: 10,
+    border: theme.palette.border.commentBorder,
+    marginBottom: 16,
+    position: "sticky",
+    top: -48,
+    zIndex: theme.zIndexes.reviewLeaderboard
+  },
+  username: {
+    width: 160,
+  },
+  karma: {
+    width: 45,
+  },
+  card: {
+    width: 400,
+    maxHeight: 400
+  },
+  showAll: {
+    color: theme.palette.primary.main,
+    marginTop: 16,
+    display: "block"
+  },
+  reviews: {
+    width: 510 
+  },
+  row: {
+    marginTop: 4
+  }
+});
+
+export const ReviewsLeaderboard = ({classes, reviews, reviewYear}: {
+  classes: ClassesType,
+  reviews?: CommentsListWithParentMetadata[],
+  reviewYear?: ReviewYear
+}) => {
+  const [truncated, setTruncated] = useState<boolean>(true)
+  const { UsersNameDisplay, Row, MetaInfo, LWTooltip, CommentsNode } = Components
+
+  const userRows = Object.entries(groupBy(reviews, (review) => review.userId))
+  const userRowsMapping = userRows.map(user => ({
+      user: user[1][0].user,
+      totalKarma: user[1].reduce((value, review) => value + review.baseScore, 0),
+      reviews: sortBy(user[1], obj => -obj.baseScore)
+  }))
+  const sortedUserRows = sortBy(userRowsMapping, obj => -obj.totalKarma)
+  const truncatedRows = truncated ? sortedUserRows.slice(0,5) : sortedUserRows
+
+  return <div className={classes.root}>
+    <Row>
+      <h3>Leaderboard</h3>
+      {reviews && <div>
+        <MetaInfo>{ reviews.length } Reviews</MetaInfo>
+        <MetaInfo>{ reviews.reduce((v, r) => v + r.baseScore, 0)} Karma</MetaInfo>
```

Comment:
minor stylistic note - i'd probably extract this kind of thing out into a variable.  benefits:

1. more self-documenting
2. makes it easier to scan the actual component structure without needing to parse business logic